### PR TITLE
[CAT-659] improve test for filtered classes

### PR DESCRIPTION
**Pull request by goatrocks**
## Summary
This adds positive and negative test cases for filtered classes, the negative being an invalid syntax.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] Jira story/task is put into PR title
- [x] Code has been self-reviewed
- [x] Hard-to-understand areas have been commented
- [ ] Documentation has been updated
- [x] My changes generate no new warnings
- [ ] Unit tests have been added for base functionality and known edge cases
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tests/integration/queries/test_workflow_component.py [hunk 1]
```diff
@@ -54,3 +54,135 @@ def test_delete_workflow_component(
...
+    
+    classes_to_filter = [
+            ["type 1"],
+            ["type 2"]
+        ]
+    filtered = AddLinkClassificationComponent(
+            workflow_id=wf.id,
+            after_component_id=after_component_id,
+            model_group_id=6108,
+            filtered_classes=[classes_to_filter],
```
Comment (meghanhickey) [inline tests/integration/queries/test_workflow_component.py:183]: Ok, so was the problem that Deloitte was passing the filtered classes as a nested list, `[["type 1", "type 1"]]` instead of `["type 1", "type 2"]`?

Comment (goatrocks) [reply tests/integration/queries/test_workflow_component.py:183]: It was actually a nested list inside a nested list [[["type 1", "type2"],["type3]]]

**Review by meghanhickey — APPROVED** [review]
